src/EventStore.ClientAPI/Projections/ProjectionDetails.cs

```diff
@@ -0,0 +1,217 @@
+    /// </summary>
+    public sealed class ProjectionDetails
+    {
+        /// <summary>
+        /// The CoreProcessingTime
+        /// </summary>
+        public readonly int CoreProcessingTime;
+
+        /// <summary>
+        /// The projection version
+        /// </summary>
+        public readonly int Version;
+
+        /// <summary>
+        /// The Epoch
+        /// </summary>
+        public readonly int Epoch;
+
+        /// <summary>
+        /// The projection EffectiveName
+        /// </summary>
+        public readonly string EffectiveName;
+
+        /// <summary>
+        /// The projection WritesInProgress
+        /// </summary>
+        public readonly int WritesInProgress;
+
+        /// <summary>
+        /// The projection ReadsInProgress
+        /// </summary>
+        public readonly int ReadsInProgress;
+
+        /// <summary>
+        /// The projection PartitionsCached
+        /// </summary>
+        public readonly int PartitionsCached;
+
+        /// <summary>
+        /// The projection Status
+        /// </summary>
+        public readonly string Status;
+
+        /// <summary>
+        /// The projection StateReason
+        /// </summary>
+        public readonly string StateReason;
+
+        /// <summary>
+        /// The projection Name
+        /// </summary>
+        public readonly string Name;
+
+        /// <summary>
+        /// The projection Mode
+        /// </summary>
+        public readonly string Mode;
+
+        /// <summary>
+        /// The projection Position
+        /// </summary>
+        public readonly string Position;
+
+        /// <summary>
+        /// The projection Progress
+        /// </summary>
+        public readonly decimal Progress;
+
+        /// <summary>
+        /// LastCheckpoint
+        /// </summary>
+        public readonly string LastCheckpoint;
+
+        /// <summary>
+        /// The projection EventsProcessedAfterRestart
+        /// </summary>
+        public readonly int EventsProcessedAfterRestart;
+
+        /// <summary>
+        /// The projection StatusUrl
+        /// </summary>
+        public readonly Uri StatusUrl;
+
+        /// <summary>
+        /// The projection StateUrl
+        /// </summary>
+        public readonly Uri StateUrl;
+
+        /// <summary>
+        /// The projection ResultUrl
+        /// </summary>
+        public readonly Uri ResultUrl;
+
+        /// <summary>
+        /// The projection QueryUrl
+        /// </summary>
+        public readonly Uri QueryUrl;
+
+        /// <summary>
+        /// The projection EnableCommandUrl
+        /// </summary>
+        public readonly Uri EnableCommandUrl;
+
+        /// <summary>
+        /// The projection DisableCommandUrl
+        /// </summary>
+        public readonly Uri DisableCommandUrl;
+
+        /// <summary>
+        /// The projection CheckpointStatus
+        /// </summary>
+        public readonly string CheckpointStatus;
```

Comment:
Agreed, I am happy with this being a string in this PR and have a separate one refactoring the CheckpointStatus.